### PR TITLE
Replace deprecated /points/lat,lon/stations api call

### PR DESCRIPTION
**Pull request by lymanepp**

### CODE REVIEW

File: pynws/nws.py [hunk 1]
```diff
@@ -59,9 +59,9 @@ def __init__(
 
     async def get_points_stations(self: Nws) -> List[str]:
```
Comment (lymanepp) [inline pynws/nws.py:60]: I didn't rename this function to avoid a breaking API change.

**Comment by MatthewFlamm** [comment]
If you prefer, you can comment "pre-commit.ci autofix" and it will auto fix formatting changes.

File: pynws/nws.py (hunk 1 above)
Comment (MatthewFlamm) [reply pynws/nws.py:60]: I think this is fine.  This function still still use the points if the gridpoints are unavailable.  Documentation is one of the weak points of this package, but you could consider briefly adding this info to the docstring.  Not a requirement.

File: pynws/simple_nws.py [hunk 2]
```diff
@@ -138,6 +138,8 @@ async def set_station(self: SimpleNWS, station: Optional[str] = None) -> None:
             if not self.stations:
                 self.stations = [self.station]
         else:
+            if not (self.wfo and self.x and self.y):
+                await self.get_points()
```
Comment (MatthewFlamm) [inline pynws/simple_nws.py]: This is already handled inside `get_points_stations`.  Is there a use case to keep it here?

**Review by MatthewFlamm — COMMENTED** [review]
Mostly LGTM.  Thanks!  I want to look at the tests too, but I have reviewed the pynws portion so far.

**Comment by MatthewFlamm** [comment]
You have made significant impact on this package, which is only furthered by this PR.  If you'd like to have write access, I would be happy to give you elevated access to this repo to better maintain it.

**Comment by lymanepp** [comment]
I'd like to look into the snapshot testing that HA is using to see if we can recapture the test data more easily in the future.

**Comment by lymanepp** [comment]
It's pretty late in the cycle for the HA 2023.9.0 release, but it would be a good idea to include this change as the integration is currently broken (for me, anyways).

**Comment by MatthewFlamm** [comment]
The HA team is usually good at adding in bug fix changes to a patch release if there is a bug report in HA.  So consider adding in an issue there.  There was another weird bug reported recently and I wonder if this is related.

**Comment by lymanepp** [comment]
I created https://github.com/home-assistant/core/issues/99475

**Comment by lymanepp** [comment]
I heard back from NWS support that this error is due to the deprecation. Were you testing in a browser or using curl? The browser will automatically redirect, but curl will not.

**Comment by MatthewFlamm** [comment]
I was previously testing in a browser.  This is good to go.  LGTM.

**Pull request merged**